tcproxy/tcproxy.go

```diff
@@ -0,0 +1,73 @@
+// Package tcproxy provides a simple wrapper around the tcproxy executable.
+// After the generic worker is refactored into engines, plugins and a runtime,
+// tcproxy will be a plugin instead.
+package tcproxy
+
+import (
+	"fmt"
+	"log"
+	"net"
+	"os/exec"
+	"strconv"
+	"time"
+
+	tcclient "github.com/taskcluster/taskcluster-client-go"
+)
+
+// TaskclusterProxy provides access to a taskcluster-proxy process running on the OS.
+type TaskclusterProxy struct {
+	command  *exec.Cmd
+	HTTPPort uint16
+	Pid      int
+}
+
+// New starts a tcproxy OS process using the executable specified, and returns
+// a *TaskclusterProxy.
+func New(taskclusterProxyExecutable string, httpPort uint16, creds *tcclient.Credentials, taskID string) (*TaskclusterProxy, error) {
```

Comment:
Any reason not to have the proxy running in the same program?  I don't think running as a different process is a bad thing, just curious.

Reply:
It is mostly about having compatibility with the existing taskcluster-proxy implementation, so when we make changes to the proxy, we can make them in the one executable, and have the workers pick up the changes. If the current implementation changed to e.g. be in rust, we could still use it. It felt less clean having IPC than it would calling a native library, but I also kind of like it now, since we can update the worker and the proxy independently from one another.